components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/dao/impl/PolicyDAOImpl.java

```diff
@@ -785,4 +794,45 @@ private static void addApplicationPolicy(Connection connection, String name, Str
             statement.execute();
         }
     }
+
+    public void deleteAPIPolicy(String policyName) throws APIMgtDAOException {
+        String sqlQuery = "DELETE FROM AM_API_POLICY WHERE NAME = ?";
+
+        try (Connection connection = DAOUtil.getConnection();
+             PreparedStatement preparedStatement = connection.prepareStatement(sqlQuery))    {
+            preparedStatement.setString(1, policyName);
+            preparedStatement.execute();
+        } catch (SQLException e)    {
+            log.error("An Error occurred while deleting Policy with name [" + policyName + "], " , e);
```

Comment:
We don't need to rollback as only one operation will be performed in one SQL connection.